node/core/approval-voting/src/criteria.rs

```diff
@@ -133,8 +183,14 @@ fn assigned_core_transcript(core_index: CoreIndex) -> Transcript {
 	t
 }
 
+fn assigned_cores_transcript(core_indices: &Vec<CoreIndex>) -> Transcript {
+	let mut t = Transcript::new(approval_types::ASSIGNED_CORE_CONTEXT);
+	core_indices.using_encoded(|s| t.append_message(b"cores", s));
```

Comment:
We do not need this commitment per se because the output defines this uniquely.

It's harmless so long as it's checked or not used.  It's maybe harmful if used unchecked by someone, but maybe it enables some optimization, like not looking up all data for the VRF input on some forwarding infrastructure.

Reply:
Thanks! I'll look into it!

Reply:
I don't know what forwarding infrastructure means in this context.

But, I am ok with removing this, it seems to make sense (AFAIU) and workable from code point of view, but tbh I have no clue if there are any security implications. It will require a bit of reordering of some checks that now will be done after checking the signature and getting the vrf outputs.

Reply:
IMO it's best from an architectural perspective to have a seperate transcript used when producing or verifying V1 vs V2 assignment VRFs. Whether that's done with an extra message or a new context defined in the `approval_types` module seems like a stylistic choice.

Reply:
https://github.com/paritytech/polkadot/pull/6802 removes redundancies for the new assignment version, but I didn't touch the VRFDelay since that would add even more complexity. I think that should be implemented in a different PR.

Reply:
Answered the security question: https://github.com/paritytech/polkadot/pull/6802#issuecomment-1448997885